Include docs about Python logs auto-instrumentation

avillela opened the pull request:
Included documentation about Python logs auto-instrumentation. Included info on the `OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED` flag, and accompanying examples. Also included usage of this flag for OTel Operator Auto-Instrumentation.

**Preview**:

- https://deploy-preview-3195--opentelemetry.netlify.app/docs/instrumentation/python/automatic/logs-example/
- https://deploy-preview-3195--opentelemetry.netlify.app/docs/kubernetes/operator/automatic/

TylerHelmuth commented on content/en/docs/kubernetes/operator/automatic.md:
````diff
@@ -310,6 +322,39 @@ in the previous step.
+        value: 'true'
+```
+
+> Note that `OTEL_LOGS_EXPORTER` must be explicitly set to `otlp_proto_http`,
````
@avillela a PR in the operator to automatically default the logs exporting behavior to http/proto like we do for traces and metrics would be awesome.

TylerHelmuth reviewed (approved):
LGTM for the operator piece.

chalin commented on content/en/docs/kubernetes/operator/automatic.md:
````diff
@@ -166,6 +168,8 @@ spec:
         value: false
 ```
 
+#### Further reading
````
Maybe:

```suggestion
#### Learn more
```

chalin commented on content/en/docs/instrumentation/python/automatic/logs-example.md:
````diff
@@ -0,0 +1,167 @@
+...
+```
+
+Note how in the sample output, the Span Event and the Log both have the same
+SpanID: `f318281c4654edc5`, since they're included under the same Span.
````
```suggestion
Note that the Span Event and the Log both have the same
SpanID (`f318281c4654edc5`). This is because they're a part of the same Span.
```

chalin reviewed (commented):
Drive-by comments, see inline.

chalin commented on content/en/docs/kubernetes/operator/automatic.md:
````diff
@@ -166,6 +168,8 @@ spec:
         value: false
 ```
 
+#### Learn more Learn more about .NET auto-instrumentation
````
Copy-paste error here: "Learn more Learn more". In any case, I'd suggest keeping the heading titles short (here and elsewhere in the page):

```suggestion
#### Learn more
```

It's clear from the line just below the heading, what the "more" is.

avillela replied:
Yes, but the markdown linter started screaming at me because the same header was duplicated for each section. If I restore to what it was, can we bypass it?

avillela commented:
@chalin I added a bit more of an explanation to the beginning of logs-example.md. If you don't mind giving it another look. 😄

chalin commented on content/en/docs/instrumentation/python/automatic/logs-example.md:
```diff
@@ -0,0 +1,167 @@
+auto-instrumentation of logs. The example below is based on the logs example in
+[OpenTelemetry Python repository][].
+
+Start by creating the examples directory and creating the example Python file:
```
```suggestion
Start by creating the examples directory and the example Python file:
```

chalin commented on content/en/docs/instrumentation/python/automatic/logs-example.md:
````diff
@@ -0,0 +1,167 @@
+touch example.py
+```
+
+Paste the following contents into `example.py`
````
```suggestion
Paste the following contents into `example.py`:
```

chalin reviewed (commented):
Two more drive-by comments -- sorry I don't have more time this week, and I'm off next week. So I'll let @cartermp et al. finish the review. Cheers!

codeboten commented on content/en/docs/instrumentation/python/automatic/logs-example.md:
```diff
@@ -0,0 +1,167 @@
+This page demonstrates how to use Python logs auto-instrumentation in
+OpenTelemetry.
+
+Unlike Traces and Metrics, there is no Logs API. There is only an SDK. For
```
Might be worth clarifying here that there *is* a logs bridge API, but it *is* different from the traces/metrics api in that it's not how application developers would be creating logs. Instead they would use this bridge API to setup log appenders in the standard logging libraries. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md

avillela replied:
Thank you! This is super helpful!!

cartermp commented:
@avillela there's a build log warning, but otherwise this looks good. Could you address it?

@open-telemetry/python-approvers can you take a look please?

avillela commented:
> @avillela there's a build log warning, but otherwise this looks good. Could you address it?
> 
> @open-telemetry/python-approvers can you take a look please?

@cartermp I fixed the build issue, but now I'm getting some errors related to a file that I didn't change, and when I try to run the same verification commands locally, I can't reproduce the errors. Any idea as to what's going on??

cartermp commented:
@avillela I'm not sure anything was fixed. The latest change has an invalid link, and there's still issues with the file format/markdownlint jobs.

cartermp commented on content/en/docs/instrumentation/python/automatic/logs-example.md:
```diff
@@ -0,0 +1,174 @@
+> Metrics API, because it's not used by application developers to create logs.
+> Instead, they would use this bridge API to setup log appenders in the standard
+> language-specific logging libraries. More information can be found
+> [here](/content-modules/opentelemetry-specification/specification//logs/bridge-api.md).
```
```suggestion
> [here](/content-modules/opentelemetry-specification/specification/logs/bridge-api.md).
```

cartermp commented:
@avillela I addressed an issue on `main` and fixed a bad link here, but there's still evidently an issue with the link:

```
docs/instrumentation/python/automatic/logs-example/index.html
  target does not exist --- docs/instrumentation/python/automatic/logs-example/index.html --> /content-modules/opentelemetry-specification/specification/logs/bridge-api.md
  ```

avillela commented:
> @avillela I addressed an issue on `main` and fixed a bad link here, but there's still evidently an issue with the link:
> 
> ```
> docs/instrumentation/python/automatic/logs-example/index.html
>   target does not exist --- docs/instrumentation/python/automatic/logs-example/index.html --> /content-modules/opentelemetry-specification/specification/logs/bridge-api.md
> ```

@cartermp finally managed to fix it. 😀
